### ISSUE TITLE
create a wallet for each user

### ISSUE THREAD

**Issue by tomislavmamic** [muqa-org/zazelenimo#2]
When a user creates an in-app account, they also get and Ethereum account and a wallet to interact with it. User can't use external wallet. User can't use wallet for anything else than what is intended within the app.

```gherkin=
Feature: Wallet
  As a user
  I want to have a wallet
  Because I want to hold currency and make transactions

  Scenario: User sees his token balance
    Given I'm not logged in
    When I open start page
    Then I can see my token balance
    
```

**Comment by b0rza** [comment]
@tomislavmamic why can't they use an external wallet?

**Comment by tomislavmamic** [comment]
> @tomislavmamic why can't they use an external wallet?

Multiple reasons:

1. For now, this will help us decrease the complexity, and simplify the UI for the user
2. We are targeting new users with this project, existing crypto users aren't our focus, but they can transfer funds from their existing wallet if they want to use this app, this will be only a small nuisance for them
3. Most of existing crypto users probably wouldn't want to have a KYC on their existing wallet for a one off donation
4. With Connect we can actually control what a user can do. Although the wallet is non-custodial, if I understood correctly, we can prevent user from accessing the wallet if we want. This is useful in cases of scam/fraud attempts.

**Comment by b0rza** [comment]
Makes sense, I was thinking about knowledgeable users. But if they are not a focus for v1 or this at all, I agree it's not needed.

**Comment by kkatusic** [comment]
@tomislavmamic if we use cometh.io wallet option, they can connect to app using existing wallet, for example MetaMask or we will disable that?

**Comment by tomislavmamic** [comment]
I suggest we disable that for now.

Tomislav Mamić

**Comment by kkatusic** [comment]
Ok
